lib/auth/methods_test.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package auth
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	apievents "github.com/gravitational/teleport/api/types/events"
+	"github.com/stretchr/testify/require"
+)
+
+type emitter struct {
+	count int
+}
+
+func (e *emitter) EmitAuditEvent(_ context.Context, event apievents.AuditEvent) error {
+	loginEvent, ok := event.(*apievents.UserLogin)
+	if ok && len(loginEvent.UserAgent) <= maxUserAgentLen {
+		e.count++
+	}
+	return nil
+}
```

Comment:
You can replace all of this with `eventstest.MockEmitter` instead.

Reply:
Done